test/integration/acceptance/fake_peer_example_test.cpp

```diff
@@ -107,3 +110,174 @@ TEST_F(FakePeerExampleFixture,
   EXPECT_TRUE(got_state_notification.load())
       << "Reached timeout waiting for MST State.";
 }
+
+/**
+ * Check that Irohad loads correct block version when having a malicious fork on
+ * the network.
+ * @given a less then 1/3 of peers having a malicious fork of the ledger
+ * @when the irohad needs to synchronize
+ * @then it refuses the malicious fork and applies the valid one
+ */
+TEST_F(FakePeerExampleFixture,
+       SynchronizerChoosesTheRightVersionOfFeorkedLedger) {
+  static constexpr size_t num_bad_peers = 3;  ///< bad fake peers
+  // the real peer is added to the bad peers as they once are failing together
+  static constexpr size_t num_peers = (num_bad_peers + 1) * 3 + 1;  ///< BFT
+  static constexpr size_t num_fake_peers = num_peers - 1;  ///< one peer is real
+
+  auto &itf = prepareState(num_fake_peers);
+
+  // let the first peers be bad
+  const std::vector<FakePeerPtr> bad_fake_peers(
+      fake_peers_.begin(), fake_peers_.begin() + num_bad_peers);
+  const std::vector<FakePeerPtr> good_fake_peers(
+      fake_peers_.begin() + num_bad_peers, fake_peers_.end());
+  const FakePeerPtr &rantipole_peer =
+      bad_fake_peers.front();  // the malicious actor
+
+  // Add two blocks to the ledger.
+  itf.sendTx(complete(baseTx(kAdminId).transferAsset(
+                          kAdminId, kUserId, kAssetId, "common_tx1", "1.0"),
+                      kAdminKeypair))
+      .skipBlock();
+  itf.sendTx(complete(baseTx(kAdminId).transferAsset(
+                          kAdminId, kUserId, kAssetId, "common_tx2", "2.0"),
+                      kAdminKeypair))
+      .skipBlock();
+
+  // Create the valid branch, supported by the good fake peers:
+  auto valid_block_storage = std::make_shared<fake_peer::BlockStorage>();
+  for (const auto &block : itf.getIrohaInstance()
+                               ->getIrohaInstance()
+                               ->getStorage()
+                               ->getBlockQuery()
+                               ->getBlocksFrom(1)) {
+    valid_block_storage->storeBlock(block);
+  }
+
+  // From now the itf peer is considered unreachable from the rest network. //
+
+  // Function to sign a block with a peer's key.
+  auto sign_block_by_peers = [](auto &&block, const auto &peers) {
```

Comment:
For now I do not see where it could be moved to. This whole file is a set of test examples - I do not expect it to include much more tests. So probably once it is needed somewhere - it will be another file and there this function can be placed.